### PR TITLE
Setup coherence

### PR DESCRIPTION
**Pull request by rhnonose**
Setup [coherence](https://github.com/smpallen99/coherence) --full-invitation type.
Setup Swoosh with Mailgun.

### CODE REVIEW

File: config/config.exs [hunk 1]
```diff
@@ -15,3 +15,17 @@ config :logger, :console,
...
+
+# %% Coherence Configuration %%   Don't remove this line
+config :coherence,
+  user_schema: CoursePlanner.User,
+  repo: CoursePlanner.Repo,
+  module: CoursePlanner,
+  logged_out_url: "/",
+  email_from_name: "Your Name",
+  email_from_email: "yourname@example.com",
+  opts: [:authenticatable, :recoverable, :lockable, :trackable, :unlockable_with_token, :invitable, :registerable]
```
Comment (ggpasqualino) [inline config/config.exs]: @ghatighorias and I were talking about the users today, and they should not be `registerable`, it should be invite only, could you update it to be like that?

File: priv/repo/migrations/20170407143340_create_course.exs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+defmodule CoursePlanner.Repo.Migrations.CreateCourse do
```
Comment (ggpasqualino) [inline priv/repo/migrations/20170407143340_create_course.exs]: Are courses necessary to setup coherence? Could you split it to another PR?

Comment (rhnonose) [inline priv/repo/migrations/20170407143340_create_course.exs]: It's important to test if it's working properly to have a custom resource.

File: config/config.exs (hunk 1 above)
Comment (rhnonose) [inline config/config.exs]: It's weird, I had the idea that --full-invitable didn't include registerable. I'll check.

File: config/config.exs [hunk 3]
```diff
@@ -15,3 +15,17 @@ config :logger, :console,
...
 import_config "#{Mix.env}.exs"
+
+# %% Coherence Configuration %%   Don't remove this line
+config :coherence,
+  user_schema: CoursePlanner.User,
+  repo: CoursePlanner.Repo,
+  module: CoursePlanner,
+  logged_out_url: "/",
+  email_from_name: "Your Name",
+  email_from_email: "yourname@example.com",
```
Comment (ggpasqualino) [inline config/config.exs]: I think it would be good to get `email_from_name` and `email_from_email` from the environment

File: priv/repo/seeds.exs [hunk 4]
```diff
@@ -1,11 +1,4 @@
...
-#     mix run priv/repo/seeds.exs
-#
-# Inside the script, you can read and write to any of your
-# repositories directly:
-#
-#     CoursePlanner.Repo.insert!(%CoursePlanner.SomeModel{})
-#
-# We recommend using the bang functions (`insert!`, `update!`
-# and so on) as they will fail if something goes wrong.
+CoursePlanner.Repo.delete_all CoursePlanner.User
```
Comment (rhnonose) [inline priv/repo/seeds.exs]: Sometimes I pollute the database with further tests, so I just run the seeds again instead of dropping and setting up the database again. I wouldn't necessarily say it's good practice, but mostly common practice.
But maybe we shouldn't have this seed at all, since it's run in every environment. Do you know any way to run separate seeds for dev/test/prod?

File: config/config.exs (hunk 3 above)
Comment (rhnonose) [inline config/config.exs]: Done.

File: priv/repo/seeds.exs (hunk 4 above)
Comment (ggpasqualino) [inline priv/repo/seeds.exs]: I am note sure what is a good practice for seeds, but we can have `priv/repo/prod_seeds.exs` and `priv/repo/dev_seeds.exs`, then we interpolate the environment into the file name when we define the `ecto.setup` alias

Comment (rhnonose) [inline priv/repo/seeds.exs]: I'm finding it difficult to separate into different files. I came up with a mix task, so it may be deemed refactoring, but according to [this issue](https://github.com/phoenixframework/phoenix_ecto/issues/8), this is a good middle-ground.

Comment (ggpasqualino) [inline priv/repo/seeds.exs]: this doesn't seem like their final word in the subject though

> For those following at home, you can do: mix run priv/repo/seeds.exs in your Phoenix app and that's all. The seeds.exs file is full of comments to tell you what to do and what not to do.

Comment (rhnonose) [inline priv/repo/seeds.exs]: Didn't find anything specifically helpful in the generated comments.

Comment (ggpasqualino) [inline priv/repo/seeds.exs]: the main point is that you don't need a task, you can run `mix run priv/repo/seeds.exs` and `seeds.exs` is a plain elixir script

I suggest we leave it as you did before, let's keep the `seeds.exs`. I will check what to do about it

Comment (rhnonose) [inline priv/repo/seeds.exs]: Alright, I'll revert it.
